subsys/bluetooth/audio/Kconfig.has

```diff
@@ -9,6 +9,7 @@ menuconfig BT_HAS
 	select EXPERIMENTAL
 	select UTF8
 	select BT_GATT_DYNAMIC_DB
+	select BT_GATT_AUTO_UPDATE_MTU
```

Comment:
Can you elaborate the Android MTU topic? Was the MTU requested too small?

Reply:
> This is an interesting addition.
> 
> We've seen issues with Android, where Android (as the central) did not properly update the MTU on connection, which mean that some notifications sent by us (as peripheral) could not be done, due to MTU not matching the requirements from e.g. BAP.

Can you elaborate? Was the MTU requested too small?

Reply:
Yes, we've seen issues where we try to send notifications on reconnect of size e.g. 30 (assuming that the MTU is 64 as mandated by e.g. the BAP spec), because the phone did not request an update to the MTU before our first notification. I think it would make sense for all services in profiles that depend on BAP (CAP, TMAP, HAP, etc.) wait with sending notifications on reconnect until after the GATT MTU has been exchanged (at least the ones where it may be larger than the initial MTU of 23). 

I think that we should probably add `select BT_GATT_AUTO_UPDATE_MTU` to any profile enabled where we act as the GATT client (e.g. `BT_HAS_CLIENT`), but we should not enable it for profiles where we are the GATT server (e.g. `BT_HAS`), as we cannot be sure that `BT_GATT_CLIENT` is enabled. 

I am unsure what happens if a Kconfig `select`s another Kconfig without satisfying the dependencies.

Reply:
> Yes, we've seen issues where we try to send notifications on reconnect of size e.g. 30 (assuming that the MTU is 64 as mandated by e.g. the BAP spec), because the phone did not request an update to the MTU before our first notification. I think it would make sense for all services in profiles that depend on BAP (CAP, TMAP, HAP, etc.) wait with sending notifications on reconnect until after the GATT MTU has been exchanged (at least the ones where it may be larger than the initial MTU of 23).

I would say it's not phone's fault that the server wants to send the notification that does not fit MTU.
On the other hand the server does not know when the client is ready to receive the notifications (e.g. the profiles or some application is ready).
As there is no dedicated L2CAP channel PSM connection, but ATT which is available right after connection this causes issues like that IMO.

> I think that we should probably add `select BT_GATT_AUTO_UPDATE_MTU` to any profile enabled where we act as the GATT client (e.g. `BT_HAS_CLIENT`), but we should not enable it for profiles where we are the GATT server (e.g. `BT_HAS`), as we cannot be sure that `BT_GATT_CLIENT` is enabled.
> 

`BT_HAS_CLIENT` selects `BT_GATT_AUTO_UPDATE_MTU` already. For profiles that are written on top of GATT Server I think that we should truncate the notifications, and apply some timeout waiting tor ATT exchange.

BTW, it would be helpful if we could have a function that will return true/false to know whether MTU was or was not already exchanged.

Reply:
> BTW, it would be helpful if we could have a function that will return true/false to know whether MTU was or was not already exchanged.

We have `bt_gatt_get_mtu`, and I guess you could simply do `bool mtu_exchanged = bt_gatt_get_mtu() == BT_ATT_DEFAULT_LE_MTU`

Reply:
Good point. 

IIRC the `bt_gatt_cb.att_mtu_updated` is even called on connection (`bt_att_connected` calls `att_chan_mtu_updated`), so we can't even use the callback. 

I'll raise a GH issue for this.

Reply:
I've created https://github.com/zephyrproject-rtos/zephyr/issues/64175 and https://github.com/zephyrproject-rtos/zephyr/issues/64172